### ISSUE TITLE
Port S.ComponentModel.* selectively

### ISSUE THREAD

**Issue by danmosemsft** [dotnet/corefx#11271]
The following members are missing and show up in the usage of the top 100 apps.

```
T:System.ComponentModel.Component
M:System.ComponentModel.Component.Dispose
M:System.ComponentModel.Win32Exception.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
T:System.ComponentModel.ReferenceConverter
T:System.ComponentModel.ComponentConverter
T:System.ComponentModel.InvalidEnumArgumentException
T:System.ComponentModel.ISynchronizeInvoke
M:System.ComponentModel.ISynchronizeInvoke.Invoke(System.Delegate,System.Object[])
T:System.ComponentModel.ListChangedEventArgs
T:System.ComponentModel.ListChangedEventHandler
M:System.ComponentModel.ListChangedEventArgs.#ctor(System.ComponentModel.ListChangedType,System.Int32)
T:System.ComponentModel.IBindingList
M:System.ComponentModel.ListChangedEventHandler.Invoke(System.Object,System.ComponentModel.ListChangedEventArgs)
M:System.ComponentModel.ToolboxItemAttribute.#ctor(System.Boolean)
T:System.ComponentModel.ToolboxItemAttribute
M:System.ComponentModel.InvalidEnumArgumentException.#ctor(System.String)
M:System.ComponentModel.InvalidEnumArgumentException.#ctor(System.String,System.Int32,System.Type)
```

Investigate and bring back as appropriate. Note that in general we only bring back types with all their members or not at all. The list seems like a grab-bag: may need to investigate what the apps are actually doing and perhaps just bring back the attributes and exceptions only.

**Comment by danmosemsft** [comment]
We need to bring back more than this as part of .net standard. Let's chat, and then please break this out.

**Comment by ewinnington** [comment]
It would be great if while looking to port S.ComponentModel, the needs of workflow foundation are considered on what they need to support [DynamicActivity in CoreWF]. (https://github.com/dmetzgar/corewf/issues/3)

**Comment by AlexGhiondea** [comment]
@ewinnington take a look at the proposed [netstandard2.0 APIs](https://github.com/dotnet/standard) and see if they would allow you to move forward.

**Comment by AlexGhiondea** [comment]
This was fixed by #12081
